src/algorithms/DominguezRios.jl

```diff
@@ -0,0 +1,267 @@
+#  Copyright 2019, Oscar Dowson and contributors
+#  This Source Code Form is subject to the terms of the Mozilla Public License,
+#  v.2.0. If a copy of the MPL was not distributed with this file, You can
+#  obtain one at http://mozilla.org/MPL/2.0/.
+
+"""
+    DominguezRios()
+
+`DominguezRios` implements the algorithm of Dominguez-Rios et al. (2021).
+"""
+
+mutable struct DominguezRios <: AbstractAlgorithm end
+
+mutable struct _Box
```

Comment:
Okay. We've got `_Rectangle` and `_Box`, now in the same namespace. Perhaps we could call this `_DominguezRiosBox` to distinguish? Or is there some commonality that we could refactor between the two algorithms?

Reply:
Yes, I am just using terms from the papers here. `_Box` has just a `priority` value attached to it. Where would `_DominguezRiosBox` live? In MOA.jl?

Reply:
It would live here, just rename the struct